docs/source/recipes/creating-a-custom-runtime-image.md

```diff
@@ -0,0 +1,148 @@
+    my-runtime-image   latest   0d1bd98fdd84        2 hours ago         887MB
+    ```
+
+1. Tag the Docker image using [`docker tag`](https://docs.docker.com/engine/reference/commandline/tag/), replacing `my-image-id`, `my-docker-id`, and `my-runtime-image` as necessary.
+
+   ```bash
+   docker tag my-image-id my-docker-id/my-runtime-image:latest
+   ```
+
+   Note: For illustrative purposes this image is tagged `latest`, which makes it the default image. If desired, replace the tag with a specific version number or identifier, such as `Vx.y.z`.
+
+1. Publish the Docker image on Docker Hub by running [`docker push`](https://docs.docker.com/engine/reference/commandline/push/), replacing `my-docker-id` and `my-runtime-image` as necessary.
+
+    ```bash
+    docker push my-docker-id/my-runtime-image
+    ```
+
+Once the image was published on Docker Hub you can [create a runtime image configuration using the Elyra UI or `elyra-metadata` CLI](/user_guide/runtime-image-conf.md) and reference the published `my-docker-id/my-runtime-image:latest` Docker image.
+
+## Additional Considerations
+
+Prior to notebook processing Elyra modifies the associated Docker container by changing the default execution command and installing additional packages. Please review the following section if
+- your Dockerfile [includes a CMD instruction](#dockerfiles-with-cmd-entries)
+- your Dockerfile [includes an ENTRYPOINT instruction](#dockerfiles-with-entrypoint-entries)
+- your package requirements [include pinned versions](#conflicting-package-dependencies)
+
+### Dockerfiles with CMD instructions
+
+If a `Dockerfile` includes a `CMD` instruction, which specifies the default command to run within the container when the container is created, you might have to customize your notebooks. When a notebook is processed as part of a pipeline the `CMD` instruction is overriden, which might have side effects. The following examples illustrate two scenarios.
+
+Scenario 1 (no undesired side-effects):
+
+The `CMD` instruction launches an application that does not need to be running when the notebook is executed. For example, the official Python Docker images might launch the interactive Python interpreter by default, like so:
+
+```bash
+...
+CMD ["python3"]
+```
+
+Notebooks will work as is because Python is automatically launched during notebook processing.
+
+Scenario 2: (undesired side-effects):
+
+The `CMD` instruction launches an application or service that a notebook consumes. For example, a Docker image might launch an application that provides computational (or connectivity) services that notebooks rely on.
+
+```bash
+...
+CMD ["python", "/path/to/application-or-service"]
+```
+
+When the Docker image is launched to process a notebook the application is unavailable because it wasn't automatically started. If feasible, the notebook could launch the application in the background in a code cell:
+
+```
+! python /path/to/application-or-service &
+```
```

Comment:
Actually this doesn't work as documented  because running as a background process isn't supported:

![image](https://user-images.githubusercontent.com/13068832/92534586-04990580-f1ea-11ea-8066-aded3914a0bb.png)

Reply:
Will have to go with 
```
import os
os.system("python hello_world.py &")
```

That appears to work as expected.

Reply:
Tested ok in a local run but fails on kfp. `hello_world.py` is a simple flask app that binds port 5000. Not sure yet what I am missing ...

![image](https://user-images.githubusercontent.com/13068832/92536660-10d39180-f1ef-11ea-9770-70a707869ce5.png)

Reply:
Turns out this was a user error. I forgot that the notebook working directory isn't the directory where the application script was located and therefore the `os.system` call never launched the app properly. 🤷